File: MatlabOctaveCodeSnippets/snip_transforms_KLT.m
N=100; %number of vectors
K=4;   %vector dimension
line=1:N*K; %straight line
noisePower=2; %noise power in Watts
temp=transpose(reshape(line,K,N)); %block signal
x=temp + sqrt(noisePower) * randn(size(temp)); %add (AWGN) noise
z=transpose(x); plot(z(:)) %prepare for plot

File: PythonCodeSnippets/snip_transforms_KLT.py
import numpy as np
from numpy.random import default_rng
import matplotlib.pyplot as plt

rng = default_rng()  # Random number generator
N = 100# Number of vectors
K = 4# Vector dimension
line = np.arange(N * K) + 1  # Straight line
noisePower = 2 # Noise power in Watts

temp = np.transpose(np.reshape(line, (K, N), order="F"))  # Block signal
x = temp + np.sqrt(noisePower) * rng.standard_normal(temp.shape) # Add (AWGN) noise

z = np.transpose(x)
plt.plot(z.ravel(order="F"))  # Prepare for plot
plt.show()
